### DISCUSSION TITLE
Checking extra fields when using cue vet

### DISCUSSION THREAD

**Discussion by cbrgm** [cue-lang/cue#1151] [Q&A] [answered]
Hello again!

I have another question that I need your help with. I want to check a file with `go vet`. I have a file `data.yaml` that I want to validate. 

__data.yaml__
```yaml
data:
  str1: "foo"
  str2: "bar"
  str3: "foobar"
```

For the validation, not only the data types are important, but also whether the sum of the characters of three attribute fields is below a certain length. For this I have a field `validStringLength` that returns `true` if the sum of the characters of the fields is below `15` characters.

__data.cue__
```cue
#Data: {
	str1: string | *""
	str2: string | *""
	str3: string | *""

	validStringLength: (len(str1)+len(str2)+len(str3) < 15)
}

data: #Data & {
    str1: "foo"
    str2: "bar"
    str3: "foobar"
}
```
[Playground Link](https://cuelang.org/play/?id=92xlWhwFB-4#cue@export@cue)

So far so good. Now I wonder if it is possible to consider another field while doing the validation? `go vet data.cue data.yaml` says the file to be validated is valid (ignoring the `validStringLength` field of course). Is there a way to take this constraint into account? Is my approach wrong and a `command` would be the right approach here?

**Comment by verdverm** [comment, marked as answer]
use `validStringLength: true & (...)` https://cuelang.org/play/?id=pUVx0h786dD#cue@export@cue

This ensures that the result of the RHS unifies with `true` and will cause an error when the RHS is false

**Reply by cbrgm** [reply]
Again, thank you so much! Spent too much time to figure that out today 😂

**Reply by myitcv** [reply]
Also to note, you can simply write the validation as a hidden definition as follows:

https://cuelang.org/play/?id=KGdf1fhxUvK#cue@export@cue

For a "proper" solution to this kind of problem see #575

**Reply by cbrgm** [reply]
Many thanks also to you for your suggestion!  @myitcv
